src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -445,104 +433,116 @@ fn check_arms<'a, 'tcx>(
     }
 }
 
-fn check_exhaustive<'p, 'a, 'tcx>(
-    cx: &mut MatchCheckCtxt<'a, 'tcx>,
+fn check_not_useful(
+    cx: &mut MatchCheckCtxt<'_, 'tcx>,
+    ty: Ty<'tcx>,
+    matrix: &Matrix<'_, 'tcx>,
+) -> Result<(), (Vec<Witness<'tcx>>, Pattern<'tcx>)> {
+    let wild_pattern = Pattern { ty, span: DUMMY_SP, kind: box PatternKind::Wild };
```

Comment:
Are `box` patterns/keyword tolerated/encouraged in the `rustc` codebase? I personally like them, but they have no path to stabilization yet, right?

Reply:
> Are `box` patterns/keyword tolerated/encouraged in the `rustc` codebase?

Well it was already there in the code so it must be, right?

> I personally like them, but they have no path to stabilization yet, right?

Right.